bundles/org.openhab.binding.vesync/src/main/java/org/openhab/binding/vesync/internal/handlers/VeSyncDeviceOutletHandler.java

```diff
@@ -0,0 +1,248 @@
+            setBackgroundPollInterval(-1);
+        } else {
+            setBackgroundPollInterval(pollRate);
+        }
+    }
+
+    @Override
+    protected void pollForDeviceData(ExpiringCache<String> cachedResponse) {
+        processV2BypassPoll(cachedResponse);
+    }
+
+    private void processV2BypassPoll(final ExpiringCache<String> cachedResponse) {
+        String responseStatus = EMPTY_STRING;
+        String responseEnergyHistory = EMPTY_STRING;
+        String responses;
+        VeSyncV2BypassOutletStatus outletStatus;
+        VeSyncV2BypassEnergyHistory energyHistory;
+        synchronized (pollLock) {
+            responses = cachedResponse.getValue();
+            boolean cachedDataUsed = responses != null;
+            if (responses == null) {
+                logger.trace("Requesting fresh response");
+                responseStatus = sendV2BypassCommand(DEVICE_GET_OUTLET_STATUS,
+                        new VeSyncRequestManagedDeviceBypassV2.EmptyPayload());
+
+                try {
+                    long end = getTimestampForToday();
+                    long start = end - 2592000; // 30 days
+                    responseEnergyHistory = sendV2BypassCommand(DEVICE_GET_ENEGERGY_HISTORY,
+                            new VeSyncRequestManagedDeviceBypassV2.GetEnergyHistory(start, end));
+                } catch (ParseException e) {
+                    logger.error("Could not parse timestamp: {}", e.getMessage());
+                }
+            } else {
+                logger.trace("Using cached response {}", responses);
+                String[] responseStrings = responses.split("?");
+                responseStatus = responseStrings[0];
+                responseEnergyHistory = responseStrings[1];
+            }
+
+            if (responseStatus.equals(EMPTY_STRING) || responseEnergyHistory.equals(EMPTY_STRING)) {
+                return;
+            }
+
+            outletStatus = VeSyncConstants.GSON.fromJson(responseStatus, VeSyncV2BypassOutletStatus.class);
+            energyHistory = VeSyncConstants.GSON.fromJson(responseEnergyHistory, VeSyncV2BypassEnergyHistory.class);
+
+            if (outletStatus == null || energyHistory == null) {
+                return;
+            }
+
+            if (!cachedDataUsed) {
+                cachedResponse.putValue(responseStatus + "?" + responseEnergyHistory);
+            }
+        }
+
+        // Bail and update the status of the thing - it will be updated to online by the next search
+        // that detects it is online.
+        if (outletStatus.isMsgDeviceOffline()) {
+            updateStatus(ThingStatus.OFFLINE);
+            return;
+        } else if (outletStatus.isMsgSuccess()) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+
+        if (!"0".equals(outletStatus.getCode())) {
+            logger.warn("Check Thing type has been set - API gave a unexpected response for an Air Purifier");
```

Comment:
Hi, sorry I know its still WIP but this warning needs updating to be for "a Outlet"

Reply:
Hi, than you for your input. You are right, i have overseen that when copy pasting your code.
You did a great job it was easy for me to extend the binding :)

Reply:
Thanks :) Glad it was easy - most of my time is lost cross referencing and running simulations to cross check other devices I don't have :)